src/uu/touch/src/touch.rs

```diff
@@ -352,18 +360,30 @@ fn parse_date(ref_time: DateTime<Local>, s: &str) -> UResult<FileTime> {
         format::YYYY_MM_DD_HH_MM,
         format::YYYYMMDDHHMM_OFFSET,
     ] {
-        if let Ok(parsed) = NaiveDateTime::parse_from_str(s, fmt) {
-            return Ok(datetime_to_filetime(&parsed.and_utc()));
+        if let Ok((parsed, modifier)) = NaiveDateTime::parse_and_remainder(s, fmt) {
+            if modifier.is_empty() {
+                return Ok(datetime_to_filetime(&parsed.and_utc()));
+            }
+            return match date_from_modifier(modifier, parsed) {
+                Ok(new_date) => Ok(datetime_to_filetime(&new_date.and_utc())),
+                Err(e) => Err(e),
+            };
```

Comment:
can probably be simplified too by something like
```suggestion
if let Ok((parsed, modifier)) = NaiveDateTime::parse_and_remainder(s, fmt) {
    return if modifier.is_empty() {
        Ok(datetime_to_filetime(&parsed.and_utc()))
    } else {
        date_from_modifier(modifier, parsed).map(|new_date| datetime_to_filetime(&new_date.and_utc()))
    };
}

```